### DISCUSSION TITLE
[help] Multiple projects on different folders

### DISCUSSION THREAD

**Discussion by llrs** [ropensci/targets#1099] [Help]
### Help

- [x] I understand and agree to <https://books.ropensci.org/targets/help.html>.

### Description

I have a pipeline with targets and now I want to create a shiny app to explore the results. For logistical reasons I don't want the shiny app to be inside the general project (I already created a quarto reveal presentation in it but it become too big and without too few reactivity).
I've been looking at issues, discussions and [the manual](https://books.ropensci.org/targets/projects.html#multiple-projects) but it is not clear what is the current best practice for that.

In https://github.com/ropensci/targets/discussions/297 I found that with untar might be what I need. But it mentions https://github.com/ropensci/targets/pull/407 which might make it easier to work with it.
I found [targets-shiny](https://github.com/wlandau/targets-shiny) but it seems like there the targets are build/written in the app folder (and doesn't make use of `tar_config_set`).

The manual suggests to use `tar_config_set`. But it seems I can change the default storage but only when I just change the storage.  Once changed, targets can load objects from the (main) project normally via `tar_load(target)`, even if I'm not in the project:

```r
library("targets")
tar_config_get("store")  # _targets
tar_config_set(store = "~/Documents/projects/a/_targets",
               config = "~/Documents/projects/a/_targets.yaml")
tar_config_get("store") # _targets
tar_config_set(store = "~/Documents/projects/a/_targets")
tar_config_get("store")  # ~/Documents/projects/a/_targets
```
The main project has no special configuration (no call to `tar_config_set`), how can I best use `tar_config_set` in the shiny app to trigger rebuilding targets iff needed?

**Comment by wlandau** [comment]
If the app only explores a well-defined subset of the results, you could use the pattern described in https://www.business-science.io/business/2020/09/24/using-drake-etl.html: write one or more [external data files](https://books.ropensci.org/targets/data.html#external-files) and track external script files for the app, which could live either inside or outside the file system of the main pipeline.

**Reply by llrs** [reply]
Yes, I already export the files and might use them to provide to the users. Thanks for confirming it.
About only being able to define one configuration at a time? Is this desired?

**Reply by wlandau** [reply]
There can only be one tar_config_set() project configuration active at a given time, but you can store many such configurations in the _targets.yaml file of the project.

**Reply by llrs** [reply]
Oh, so it cannot reuse the configuration of a different path? I think that was my misunderstanding. Thanks!

**Reply by wlandau** [reply]
`tar_config_set()` and `tar_config_get()` by default access or modify the settings of the active project, and you can switch projects either with the `project` argument or the `TAR_PROJECT` environment variable: https://books.ropensci.org/targets/projects.html#multiple-projects
